### ISSUE TITLE
A more customizable  REPL prompt

### ISSUE THREAD

**Issue by cormullion** [JuliaLang/julia#20692]
It's currently possible to set the REPL prompt:

```
julia> Base.active_repl.interface.modes[1].prompt="v$(VERSION.minor)> "

v5> 

```
But it would be useful if you could set the REPL prompt so that it always displays up to date information. (The Julia version obviously won't change, but other things do.)

```
julia> Base.active_repl.interface.modes[1].prompt="$(pwd())> ";

/private/tmp> 
shell> cd ~/.julia/v0.5/
/Users/me/.julia/v0.5

/private/tmp>                           # prompt not updated
```

Or: 

```
julia> Base.active_repl.interface.modes[1].prompt="$(Dates.format(now(), "HH:MM:SS"))> "
"14:01:01> "

14:01:01> 

14:01:01>                               # prompt not updated
```

If the REPL evaluated a current prompt string anew each time, that would work. Perhaps the prompt could be an environment variable? Or the REPL could provide a "run this function after every evaluation just before you see the "julia> " prompt again" function?

(Why no PR? Well, I looked through REPL.jl, but couldn't understand it well enough to work out how to get the prompt string evaluated each time.)

(Here's a [MatLab](http://undocumentedmatlab.com/blog/setprompt-setting-matlab-desktop-prompt) equivalent.)

**Comment by Keno** [comment]
The repl has some support for this. See e.g. https://github.com/Keno/TerminalExtensions.jl/blob/master/src/TerminalExtensions.jl#L143-L150

**Comment by KristofferC** [comment]
```julia
if !isdefined(repl,:interface)
    repl.interface = Base.REPL.setup_interface(repl)
end
```

wow.. that is sneaky and clever ;)

**Comment by Keno** [comment]
Well, it's not particularly clever, since I added that hack to the REPL way back when (https://github.com/JuliaLang/julia/blob/master/base/REPL.jl#L929-L933) such that that would work

**Comment by rfourquet** [comment]
Fixed in #22809. Of course, a nicer way to set the prompt is still wanted (I will fix that eventually if no-one beats me to it).
